blockchain/upgrade.go

```diff
@@ -3505,7 +3505,7 @@ func upgradeSpendJournal(ctx context.Context, b *BlockChain) error {
 //
 // NOTE: The database info housed by the passed blockchain instance will be
 // updated with the latest versions.
-func upgradeUtxoDb(ctx context.Context, b *BlockChain) error {
+func upgradeUtxoDb(ctx context.Context, b *BlockChain, utxoDb database.DB) error {
```

Comment:
Similarly, this doesn't appear to really need access to the entire chain anymore.  Only the block database, `UtxoCache`, and version details down in `upgradeUtxoSetToVersion3`.

I know this probably doesn't matter too much, but I think it'd be nice to avoid passing ` BlockChain` anywhere down into the db code in general to avoid the temptation to use other things on it that lead to upgrade issues due to ordering.  I noticed it back when reviewing the PR that introduced all this, but I didn't say anything since it previously was a lot more tightly integrated, but given this is improving the overall modularity, I figured I'd call it out here.

Reply:
Agreed, made an update along these lines to pass in `utxoBackend *LevelDbUtxoBackend` so that the migrations can use the backend or underlying database where needed.  Once the new UTXO backend is brought in, this could likely be updated to pass in the `UtxoBackend` interface rather than the concrete type.